python_modules/dagster/dagster/_core/telemetry.py

```diff
@@ -523,6 +529,32 @@ def log_action(
         )
 
 
+def log_step_event(event: DagsterEvent, pipeline_context: PlanOrchestrationContext):
```

Comment:
nit: make this `log_dagster_event` or something, potentially we might want to collect information from non-step events in the future (asset stuff for example) and this is a good centralized entrypoint